### PR TITLE
Update project dependencies

### PR DESCRIPTION
**Pull request by jocgir**
Most dependencies in that project were deprecated.

### CODE REVIEW

File: docker.go [hunk 1]
```diff
@@ -116,6 +120,7 @@ func (docker *dockerConfig) call() int {
 			os.Mkdir(temp, 0755)
 		}
 		dockerArgs = append(dockerArgs, "-v", fmt.Sprintf("%s%s:/var/tgf", convertDrive(tempDrive), tempFolder))
+		config.Environment["TGF_TEMP_FOLDER"] = path.Join(tempDrive, tempFolder)
```
Comment (jocgir) [inline docker.go:123]: This export is used to publish the name of the temporary folder on the host used to execute the actual command.

This is used within the bootstrap to display the complete path relative to the host when entering in a shell `tgf bash`. Then, it is easy to open an editor (VSCode or else) directly in the temporary folder.

File: README.md [hunk 2]
```diff
@@ -207,7 +207,7 @@ Flags:
   -E, --entrypoint=terragrunt   Override the entry point for docker
       --current-version         Get current version information
       --all-versions            Get versions of TGF & all others underlying utilities
-  -L, --logging-level=<level>   Set the logging level (critical=0, error=1, warning=2, notice=3, info=4, debug=5, full=6)
+  -L, --logging-level=<level>   Set the logging level (panic=0, fatal=1, error=2, warning=3, info=4, debug=5, trace=6)
```
Comment (jocgir) [inline README.md:210]: The help string has never been adjusted since we moved to logrus.

File: docker.go [hunk 3]
```diff
@@ -45,9 +48,10 @@ func (docker *dockerConfig) call() int {
 	// Change the default log level for terragrunt
 	const logLevelArg = "--terragrunt-logging-level"
 	if !util.ListContainsElement(command, logLevelArg) && filepath.Base(config.EntryPoint) == "terragrunt" {
-		if config.LogLevel == "6" || strings.ToLower(config.LogLevel) == "full" {
-			config.LogLevel = "debug"
-			config.Environment["TF_LOG"] = "DEBUG"
+		level, _ := strconv.Atoi(config.LogLevel)
```
Comment (ycanty) [inline docker.go:51]: Should we also set a more proper default if LogLevel is wrong, like "abc"?  With the current code that will silently set the level to 0 (panic) so almost no logs...  Is that what we want?  Should we default to something more verbose to like "info" in this case?

File: go.mod [hunk 4]
```diff
@@ -1,24 +1,23 @@
 module github.com/coveooss/tgf
 
-go 1.12
-
-replace github.com/gruntwork-io/terragrunt => github.com/coveooss/terragrunt v1.4.0
+go 1.13
```
Comment (ycanty) [inline go.mod:3]: Why not upgrade to go 1.14?

Comment (jocgir) [reply go.mod:3]: In fact, I don't remember changing this line. But the go version specified here does not specify the version used to compile the code, but the minimum requirement if this module is imported from elsewhere. If I put 1.14 here, I will have to update all projects including this project (which is none in that specific case 😀 )

File: docker.go (hunk 3 above)
Comment (jocgir) [reply docker.go:51]: You are right that we should add a warning, but the effect of a wrong level name here is just to ignore it. The goal is just to trap special cases where the logging level is expressed as a number and is greater than 6 or set to full. If it is not the case, the logging level will simply be passed to terragrunt and should be managed there. In fact, there is actually no real logging implemented in tgf.

I am working on another PR where I implement a more intelligent logging directly in tgf.

**Pull request merged**